sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/OkHttpHttpClient.java

```diff
@@ -156,18 +155,18 @@ private static HttpResponse toHttpResponse(HttpRequest request, okhttp3.Response
                 body.close();
             }
 
-            return new OkHttpBufferedResponse(response, request, EMPTY_BODY, eagerlyConvertHeaders);
+            return new OkHttpBufferedResponse(response, request, eagerlyConvertHeaders, EMPTY_BODY);
         }*/
 
         // Use a buffered response when we are eagerly reading the response from the network and the body isn't empty.
         if (eagerlyReadResponse || ignoreResponseBody) {
             try (ResponseBody body = response.body()) {
                 byte[] bytes = (body != null) ? body.bytes() : EMPTY_BODY;
 
-                return new OkHttpBufferedResponse(response, request, bytes, eagerlyConvertHeaders);
+                return new OkHttpResponse(response, request, eagerlyConvertHeaders, bytes);
```

Comment:
We merged both OkHttp response types and handle buffering internally when necessary.

Reply:
To be clearer: the response itself does not know whether to buffer or not, we create the instance with the buffered body bytes in a `BinaryData` if either `eagerlyReadResponse` or `ignoreResponseBody` are set to `true`, like what we do on Azure Core. Otherwise, the `BinaryData` holds an `InputStream` we get from the OkHttp `ResponseBody`.